packages/synapse-react-client/src/synapse-queries/entity/useEntity.ts

```diff
@@ -370,23 +372,34 @@ export function useUpdateEntityACL(
   )
 }
 
-export function useUpdateTable(
+type UpdateTableMutationRequest = {
+  entityId: string
+  originalColumnModels: ColumnModel[]
+  newColumnModels: SetOptional<ColumnModel, 'id'>[]
+}
+
+export function useUpdateTableColumns(
   options?: UseMutationOptions<
     unknown,
     SynapseClientError,
-    TableUpdateTransactionRequest
+    UpdateTableMutationRequest
   >,
 ) {
   const queryClient = useQueryClient()
   const { accessToken, keyFactory } = useSynapseContext()
 
-  return useMutation<
-    unknown,
-    SynapseClientError,
-    TableUpdateTransactionRequest
-  >(
-    (request: TableUpdateTransactionRequest) =>
-      SynapseClient.updateTable(request, accessToken),
+  return useMutation<unknown, SynapseClientError, UpdateTableMutationRequest>(
+    async (request: UpdateTableMutationRequest) => {
+      // This call will create new column models as appropriate
+      const transactionRequest = await createTableUpdateTransactionRequest(
+        accessToken!,
+        request.entityId,
+        request.originalColumnModels,
+        request.newColumnModels,
+      )
+
+      return SynapseClient.updateTable(transactionRequest, accessToken)
+    },
```

Comment:
Before the refactor, the caller of this mutation had to 
1. Create column models using `createTableUpdateTransactionRequest`
2. Call `mutate`, which called `updateTable`.

I refactored so the mutation also creates the update, because this is always going to be an atomic step. We can handle loading/error states consistently while we wait for this sequence of requests to be made.